plugins/utils/rest.nix

```
@@ -434,9 +381,60 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
       }
     ];
 
+    # TODO: introduced 2024-09-23: remove after 24.11
+    warnings =
+      let
+        definedOpts = lib.filter (opt: lib.hasAttrByPath (lib.toList opt) cfg.settings) [
+          "client"
+          "env_file"
+          "env_pattern"
+          "env_edit_command"
+          "encode_url"
+          "skip_ssl_verification"
+          [
+            "logs"
+            "level"
+          ]
+          [
+            "logs"
+            "save"
+          ]
+          [
+            "result"
+            "split"
+          ]
+          [
+            "result"
+            "behavior"
+          ]
+          [
+            "result"
+            "statistics"
+          ]
+          [
+            "result"
+            "decode_url"
+          ]
+          [
+            "result"
+            "formatters"
+          ]
+          [
+            "result"
+            "keybinds"
+          ]
+        ];
+      in
+      lib.optional (definedOpts != [ ]) ''
+                Nixvim(plugins.rest): The following v2 settings options are no longer supported in v3:
+        ${lib.concatMapStringsSep "\n" (opt: "  - ${lib.showOption (lib.toList opt)}") definedOpts}
+      '';
```

Sorry to comment on formatting, but the indentation seems messed up:
```suggestion
      lib.optional (definedOpts != [ ]) ''
        Nixvim(plugins.rest): The following v2 settings options are no longer supported in v3:
        ${lib.concatMapStringsSep "\n" (opt: "  - ${lib.showOption (lib.toList opt)}") definedOpts}
      '';
```